iroh/src/magicsock/relay_actor.rs

```diff
@@ -676,75 +669,41 @@ impl RelayActor {
         self.log_active_relay();
     }
 
-    /// Cleans up stale [`ActiveRelayActor`]s.
-    ///
-    /// This not only checks if the relays have been used recently, but also makes sure that
-    /// all relay actors are running.  In particular this is called whenever an
-    /// [`ActiveRelayActor`] task finishes.
-    async fn clean_stale_relay(&mut self) {
-        trace!("checking {} relays for staleness", self.active_relays.len());
-        let now = Instant::now();
-
-        // Futures who return Some(RelayUrl) if the relay needs to be cleaned up.
-        let check_futs = self.active_relays.iter().map(|(url, handle)| async move {
-            let (tx, rx) = oneshot::channel();
-            handle
-                .inbox_addr
-                .send(ActiveRelayMessage::GetLastWrite(tx))
-                .await
-                .ok();
-            match rx.await {
-                Ok(last_write) if last_write.duration_since(now) <= RELAY_INACTIVE_CLEANUP_TIME => {
-                    None
-                }
-                _ => Some(url.clone()),
+    /// Cleans up [`ActiveRelayActor`]s which have stopped running.
+    fn reap_active_relays(&mut self) {
+        let mut stopped_actors = Vec::with_capacity(self.active_relays.len());
```

Comment:
you don't have any `await`s in here, so I think you could just use `self.active_relays.retain`

Reply:
whoa, retain is nice!